### PR TITLE
[CI improvement] Move dataset_tests to an independent directory

### PR DESCRIPTION
**Pull request by caojy1998**
## Description

Currently, some tests which only focuses on datasets validation are time consuming. We try to remove them form our CI test, which mostly focuses on unittest. And we will later use the regression system to run all these dataset tests daily.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/test_integrate/data_test.py
Comment (frozenbugs) [inline tests/test_integrate/data_test.py:1]: tests/integration/test_data.py

**Review by frozenbugs — APPROVED** [review]

File: tests/test_integrate/data_test.py
Comment (caojy1998) [reply tests/test_integrate/data_test.py:1]: OK, I have changed the name of file and directory.

File: tests/integration/test_data.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import pytest
+import yaml
+from dgl import DGLError
+
+
+@unittest.skipIf(
+    F._default_context_str == "gpu",
+    reason="Datasets don't need to be tested on GPU.",
+)
+@unittest.skipIf(dgl.backend.backend_name == "mxnet", reason="Skip MXNet")
```
Comment (frozenbugs) [inline tests/integration/test_data.py:23]: Please double check whether pytest is a good option to run integration test like this.

**Comment by frozenbugs** [comment]
Also remove from the original code?

**Comment by caojy1998** [comment]
> Also remove from the original code?

Yes, I am still editing the code

**Review by frozenbugs — APPROVED** [review]

File: tests/integration/test_data.py (hunk 1 above)
Comment (caojy1998) [reply tests/integration/test_data.py:23]: OK, I will later figure it out on DGL_scripts repo to see whether asv_run is a good choice.

**Pull request merged**